magicblock-committor-service/src/intent_executor/intent_executor.rs

```diff
@@ -0,0 +1,247 @@
+        match &mut prepared_message {
+            VersionedMessage::V0(value) => {
+                value.recent_blockhash = latest_blockhash;
+            }
+            VersionedMessage::Legacy(value) => {
+                warn!("TransactionPreparator v1 does not use Legacy message");
+                value.recent_blockhash = latest_blockhash;
+            }
+        };
+
+        let transaction =
+            VersionedTransaction::try_new(prepared_message, &[&self.authority])
+                .map_err(|err| (err.into(), None))?;
+        let result = self
+            .rpc_client
+            .send_transaction(
+                &transaction,
+                &MagicBlockSendTransactionConfig::ensure_committed(),
+            )
+            .await
+            .map_err(|err| {
+                let signature = err.signature();
+                (err.into(), signature)
+            })?;
+
+        Ok(result.into_signature())
+    }
+
+    fn persist_result<P: IntentPersister>(
+        persistor: &Option<P>,
+        result: &IntentExecutorResult<ExecutionOutput>,
+        message_id: u64,
+        pubkeys: &[Pubkey],
+    ) {
+        match result {
+            Ok(value) => {
+                let signatures = CommitStatusSignatures {
+                    process_signature: value.commit_signature,
+                    finalize_signature: Some(value.commit_signature)
+                };
+                let update_status = CommitStatus::Succeeded(signatures);
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::EmptyIntentError) => {
+                let update_status = CommitStatus::Failed;
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::FailedToFitError)) => {
+                let update_status = CommitStatus::PartOfTooLargeBundleToProcess;
+                persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+            }
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::TaskBuilderError(err))) => {
+                match err {
+                    crate::tasks::task_builder::Error::CommitTasksBuildError(_) => {
+                        let update_status = CommitStatus::Failed;
+                        persist_status_update_by_message_set(persistor, message_id, pubkeys, update_status);
+                    }
+                    crate::tasks::task_builder::Error::FinalizedTasksBuildError(_) => {
+                        // TODO: commit signature to set this
+                    }
+                 }
+            },
+            Err(Error::FailedCommitPreparationError(crate::transaction_preperator::error::Error::DeliveryPreparationError(_))) => {
+                // Persisted internally
```

Comment:
What does that mean? How is the error communicated?

Reply:
This method it `persist_result` - just updates db once intent execution is finished. Result is propagated further

> // Persisted internally

Refers to a fact that DeliveryPreparator updates Persister itself, so we can skip it here

Reply:
When I ask those questions I mean to add a comment explaining exactly that so that other devs looking at the code have the same info. 
The current comment is just way to short and doesn't give enough info IMO